### ISSUE TITLE
Moment is crash after run hours

### ISSUE THREAD

**Issue by RafaelSalomao1981** [moment/moment#4166]
Hi, in my chat application in nodejs, after hours i receveid this error.

This crash my application.

I tried debugging the moment code, but I did not succeed.

<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x6dd3408cfb51 <JS Object>
    1: createLocalOrUTC(aka createLocalOrUTC) [/root/node/node_modules/live_webcam/node_modules/moment/moment.js:2577] [pc=0x6dd345ba109c] (this=0x6dd340804381 <undefined>,input=0x6dd340804381 <undefined>,format=0x6dd340804381 <undefined>,locale=0x6dd340804381 <undefined>,strict=0x6dd340804381 <undefined>,isUTC=0x6dd340804271 <false>)
    2: createLocal(aka createLocal) [/root/node/node_modules...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node::Abort() [node]
 2: 0x109838c [node]
 3: v8::Utils::ReportApiFailure(char const*, char const*) [node]
 4: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [node]
 5: v8::internal::Factory::NewTransitionArray(int) [node]
 6: v8::internal::TransitionArray::Insert(v8::internal::Handle<v8::internal::Map>, v8::internal::Handle<v8::internal::Name>, v8::internal::Handle<v8::internal::Map>, v8::internal::SimpleTransitionFlag) [node]
 7: v8::internal::Map::CopyReplaceDescriptors(v8::internal::Handle<v8::internal::Map>, v8::internal::Handle<v8::internal::DescriptorArray>, v8::internal::Handle<v8::internal::LayoutDescriptor>, v8::internal::TransitionFlag, v8::internal::MaybeHandle<v8::internal::Name>, char const*, v8::internal::SimpleTransitionFlag) [node]
 8: v8::internal::Map::CopyAddDescriptor(v8::internal::Handle<v8::internal::Map>, v8::internal::Descriptor*, v8::internal::TransitionFlag) [node]
 9: v8::internal::Map::CopyWithField(v8::internal::Handle<v8::internal::Map>, v8::internal::Handle<v8::internal::Name>, v8::internal::Handle<v8::internal::FieldType>, v8::internal::PropertyAttributes, v8::internal::Representation, v8::internal::TransitionFlag) [node]
10: v8::internal::Map::TransitionToDataProperty(v8::internal::Handle<v8::internal::Map>, v8::internal::Handle<v8::internal::Name>, v8::internal::Handle<v8::internal::Object>, v8::internal::PropertyAttributes, v8::internal::Object::StoreFromKeyed) [node]
11: v8::internal::LookupIterator::PrepareTransitionToDataProperty(v8::internal::Handle<v8::internal::JSObject>, v8::internal::Handle<v8::internal::Object>, v8::internal::PropertyAttributes, v8::internal::Object::StoreFromKeyed) [node]
12: v8::internal::StoreIC::LookupForWrite(v8::internal::LookupIterator*, v8::internal::Handle<v8::internal::Object>, v8::internal::Object::StoreFromKeyed) [node]
13: v8::internal::StoreIC::UpdateCaches(v8::internal::LookupIterator*, v8::internal::Handle<v8::internal::Object>, v8::internal::Object::StoreFromKeyed) [node]
14: v8::internal::StoreIC::Store(v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Name>, v8::internal::Handle<v8::internal::Object>, v8::internal::Object::StoreFromKeyed) [node]
15: v8::internal::Runtime_StoreIC_Miss(int, v8::internal::Object**, v8::internal::Isolate*) [node]
16: 0x6dd3410092a7
Aborted

How i can fixed it?

**Comment by icambron** [comment]
That's an out of memory error. Just because it crashed trying to allocate memory for a Moment object doesn't mean Moment is the one leaking your memory. It's more likely that you're leaking it. You might even be leaking it with Moment objects! You'd need to produce a small example to show that. Then maybe we could help.

**Comment by rennanods** [comment]
Hello,

I'm experiencing the same problem, but it does not occur running localhost (macOS), only on the server (linux). The problem only occurs when I use any moment.js service to format the date "2018-10-21", only this date, any other date works normally. I have already compared the node and moment.js versions, they are the same as the ones I use in localhost. I also tried using other versions of node and moment.js, but it does not work. I tried formatting this date using Date () and it worked, but I use the moment in many places of the source code and I did not want to stop using this library.

Anyone have any idea what it might be?
